src/app/components/drawings-listeners/link-created/link-created.component.ts

```diff
@@ -35,12 +35,45 @@ export class LinkCreatedComponent implements OnInit, OnDestroy {
   }
 
   onLinkCreated(linkCreated: MapLinkCreated) {
+    const xLength = Math.abs(linkCreated.sourceNode.x - linkCreated.targetNode.x);
+    const yLength = Math.abs(linkCreated.sourceNode.y - linkCreated.targetNode.y);
+    const zLength = Math.sqrt(Math.pow(xLength, 2) + Math.pow(yLength, 2));
+    const x = (45 / zLength) * xLength;
+    const y = (45 / zLength) * yLength;
+
+    let xLabelSourceNode = 0;
+    let yLabelSourceNode = 0;
+    let xLabelTargetNode = 0;
+    let yLabelTargetNode = 0;
+
+    if ((linkCreated.sourceNode.x <= linkCreated.targetNode.x) && (linkCreated.sourceNode.y <= linkCreated.targetNode.y)) {
+      xLabelSourceNode = Math.floor(linkCreated.sourceNode.width/2) + Math.round(x) + 5;
+      yLabelSourceNode = Math.floor(linkCreated.sourceNode.height/2) + Math.round(y) + 5;
+      xLabelTargetNode = Math.floor(linkCreated.targetNode.width/2) - Math.round(x) - 15;
```

Comment:
These values (5, 15) should moved to constants with other, I prefer to do that in another pr

Reply:
@piotrpekala7 makes sens, thanks